File: 1.1/src/c/cz/mg/vulkan/PFN_vkResetFences.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

jint Java_cz_mg_vulkan_PFNvkResetFences_callNative(JNIEnv* env, jclass jc, jlong address, jlong device, jint fenceCount, jlong pFences){
    (void)env;
    (void)jc;
    PFN_vkResetFences f = (PFN_vkResetFences)jniLongToFunctionPointer(address);
    return f(
        *((VkDevice*)jniLongToPointer(device)),
        (uint32_t)fenceCount,
        ((VkFence*)jniLongToPointer(pFences))
    );
}

File: 1.1/src/cz/mg/vulkan/PFNvkResetFences.java
package cz.mg.vulkan;

public class PFNvkResetFences extends VkFunctionPointer {
    public PFNvkResetFences() {
    }

    protected PFNvkResetFences(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkResetFences(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkResetFences(long value) {
        setValue(value);
    }

    public PFNvkResetFences(VkInstance instance) {
        super(instance, new VkString("vkResetFences"));
    }

    public int call(VkDevice device, int fenceCount, VkFence pFences){
        return callNative(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddressNative(), fenceCount, pFences != null ? pFences.getVkAddress() : VkPointer.NULL);
    }

    protected static native int callNative(long vkaddress, long device, int fenceCount, long pFences);

}
